.changelog/15297.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+all: Remove the use of the deprecated ioutil package throughout the codebase in favour of io and os packages which are still maintained and also introduce a lint rule which forbids the use of ioutil in commits.
```

Comment:
Many times this sort of fixup doesn't necessitate a changelog entry. Your call if you think we should have one.

Reply:
Makes sense! I was on the fence about it, so I'll remove it then, I think the commit message should be good enough!